### PR TITLE
Add Async Rest Client

### PR DESCRIPTION
**Pull request by macwilk**
## Overview
- Rerun OpenAPI code generation pre-async swap
- Generate OpenAPI code post-async swap
    - Converts the existing RESTful client to an `aiohttp` based async version
- Implements an Async Wrapper for the openapi-generated client
- Implements a Sync wrapper for the Async Wrapper to keep backwards compatibility
    - Note: this was super annoying to get working so please let me know if the workarounds look too terrible
- Adds example of using the async RESTful Client
- Adds a unit test
## Dep Changes
- +aiohttp
    - required for the async RESTful client
- +aiohttp-retry
    - required for the async RESTful client
- asyncio-atexit
    - the aiohttp openapi-generated client doesn't auto close its connection so this is used to prevent errors when the main thread closes down (gracefully terminating the async connection)

### CODE REVIEW

File: examples/api/async_api.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from hatchet_sdk import Hatchet, WorkflowList
+
+load_dotenv()
+
+hatchet = Hatchet(debug=True)
+
+
+async def main():
+    hatchet_async_rest_client = await hatchet.get_async_rest_client()
```
Comment (grutt) [inline examples/api/async_api.py:13]: i think we have hatchet.admin and hatchet.aio.admin can you look into that pattern and can we do the same for this and instantiate the aio client on first get maybe?

**Review by grutt — CHANGES_REQUESTED** [review]

File: examples/api/async_api.py (hunk 1 above)
Comment (grutt) [reply examples/api/async_api.py:13]: also it looks like we're replacing the sync api? is this a breaking change?

Comment (macwilk) [reply examples/api/async_api.py:13]: not a breaking change unless someone is doing something fucky with the library that involves touching the underlying objects on the `RestApi` object.


I completely forgot about `hatchet.aio`, let me see if this'll work. The main issue is that `aiohttp` requires an active event loop or it explodes so layering it on our existing hatchet client was looking pretty terrible and so I opted for the current design.

File: hatchet_sdk/clients/rest/configuration.py [hunk 2]
```diff
@@ -196,12 +195,9 @@ def __init__(
            Set this to the SNI value expected by the server.
         """
 
-        self.connection_pool_maxsize = multiprocessing.cpu_count() * 5
-        """urllib3 connection pool's maximum number of connections saved
-           per pool. urllib3 uses 1 connection as default value, but this is
-           not the best value when you are making a lot of possibly parallel
-           requests to the same host, which is often the case here.
-           cpu_count * 5 is used as default value to increase performance.
+        self.connection_pool_maxsize = 100
```
[the remark this answers is not in the document]
Comment (macwilk) [reply hatchet_sdk/clients/rest/configuration.py:198]: this comes from the openapi generated stuffs, assuming it is?

Do you have any thoughts on what this value should be?

[the remark this answers is not in the document]
Comment (grutt) [reply hatchet_sdk/clients/rest/configuration.py:198]: no meaningful thoughts

**Pull request merged**